### PR TITLE
chore(modules/playback-engine/html5-output) - update inner API

### PR DESCRIPTION
**Pull request by maxvektor**
First part of https://github.com/wix/playable/pull/103 broken into pieces

### CODE REVIEW

File: src/modules/playback-engine/types.ts [hunk 1]
```diff
@@ -100,19 +96,30 @@ interface IPlaybackEngine {
   getDebugInfo(): IEngineDebugInfo;
 
   destroy(): void;
+
+  changeOutput(chromecastOutput: IVideoOutput): void;
+  resetOutput(): void;
+}
+
+interface IEngineDebugInfo {
```
Comment (BrooklynKing) [inline src/modules/playback-engine/types.ts]: Add `[key: string]: any` maybe?

**Comment by maxvektor** [comment]
I have removed redundant start time from setSrc.

File: src/modules/playback-engine/output/native/adapters/native.ts [hunk 2]
```diff
@@ -42,10 +42,6 @@ export default function getNativeAdapterCreator(
       this._bindCallbacks();
     }
 
-    get currentUrl() {
```
Comment (BrooklynKing) [inline src/modules/playback-engine/output/native/adapters/native.ts]: Why you removing it?

Comment (maxvektor) [reply src/modules/playback-engine/output/native/adapters/native.ts]: Editor shows no usage of this property.

Comment (maxvektor) [reply src/modules/playback-engine/output/native/adapters/native.ts]: take a look on a second commit, pls. I removed _curentSrc from engine. Now we take it directly from output(adapters). Looks like this approach works fine, but still validate me on this pls.

File: src/modules/playback-engine/output/native/html5video-output.ts [hunk 3]
```diff
@@ -300,31 +296,23 @@ export default class NativeOutput implements IVideoOutput {
     return this._video.videoWidth;
   }
 
-  get error() {
-    return this._video.error;
-  }
-
   get src() {
+    if (this.attachedAdapter) {
```
Comment (maxvektor) [inline src/modules/playback-engine/output/native/html5video-output.ts]: no if

File: src/modules/playback-engine/output/native/state-engine.ts [hunk 4]
```diff
@@ -177,5 +177,7 @@ export default class StateEngine {
 
   destroy() {
     this._unbindEvents();
+
+    this._eventEmitter = null;
```
Comment (maxvektor) [inline src/modules/playback-engine/output/native/state-engine.ts]: remove

**Pull request closed without merging**